File: TBAnnotationClustering/TBClusterAnnotation.m
//
//  TBClusterAnnotation.m
//  TBAnnotationClustering
//
//  Created by Theodore Calmes on 10/8/13.
//  Copyright (c) 2013 Theodore Calmes. All rights reserved.
//

#import "TBClusterAnnotation.h"

@implementation TBClusterAnnotation

- (id)initWithCoordinate:(CLLocationCoordinate2D)coordinate count:(NSInteger)count
{
    self = [super init];
    if (self) {
        _coordinate = coordinate;
        _title = [NSString stringWithFormat:@"%ld hotels in this area", (long)count];
        _count = count;
    }
    return self;
}

- (NSUInteger)hash
{
    NSString *toHash = [NSString stringWithFormat:@"%.5F%.5F", self.coordinate.latitude, self.coordinate.longitude];
    return [toHash hash];
}

- (BOOL)isEqual:(id)object
{
    return [self hash] == [object hash];
}

@end

File: TBAnnotationClustering-Swift/TBClusterAnnotation.swift
//
//  TBClusterAllocation.swift
//  TBAnnotationClustering-Swift
//
//  Created by Eyal Darshan on 1/1/16.
//  Copyright © 2016 eyaldar. All rights reserved.
//

import MapKit

class TBClusterAnnotation: NSObject, MKAnnotation {
    
    let coordinate: CLLocationCoordinate2D
    let count:Int

    var title:String?
    var subtitle:String?
    
    init(coordinate:CLLocationCoordinate2D, count:Int) {
        self.coordinate = coordinate
        self.title = "\(count) hotels in this area"
        self.count = count
        
        super.init()
    }
    
    override func isEqual(object: AnyObject?) -> Bool {
        if let other = object as? TBClusterAnnotation {
            return coordinate.longitude == other.coordinate.longitude &&
                    coordinate.latitude == other.coordinate.latitude
        }
        
        return false
    }
}
